src/Uno.UWP/Storage/FileIO.cs

```diff
@@ -0,0 +1,224 @@
+		{
+			StringBuilder output = new StringBuilder(); 
+			foreach (string line in lines)
+			{
+				output.Append(line + Environment.NewLine);
+			}
+
+			return output.ToString();
+		}
+
+		public static IAsyncAction WriteLinesAsync(IStorageFile file, IEnumerable<string> lines, Streams.UnicodeEncoding encoding)
+			=> WriteTextAsync(file, ConvertLinesToString(lines), encoding);
+
+		public static IAsyncAction AppendLinesAsync(IStorageFile file, IEnumerable<string> lines, Streams.UnicodeEncoding encoding)
+			=> AppendTextAsync(file, ConvertLinesToString(lines), encoding);
+
+
+		private static async Task AppendWriteTextAsync(IStorageFile file, string contents, bool append)
+		{
+			Streams.UnicodeEncoding unicodeEncoding = Streams.UnicodeEncoding.Utf8;
+
+			using (Stream fileReadStream = await file.OpenStreamForReadAsync())
+			{
+				using (BinaryReader binReader = new BinaryReader(fileReadStream))
+				{
+					try
+					{
+						byte firstByte = binReader.ReadByte();
+						switch (firstByte)
+						{
+							case 0xfe:
+								unicodeEncoding = Streams.UnicodeEncoding.Utf16BE;
+								break;
+							case 0xff:
+								unicodeEncoding = Streams.UnicodeEncoding.Utf16LE;
+								break;
+						}
+					}
+					catch
+					{
+						// probably file is empty
+					}
+
+				}
+			}
+
+			AppendWriteTextAsync(file, contents, unicodeEncoding, append);
+		}
+
+		public static IAsyncAction WriteTextAsync(IStorageFile file, string contents) => AppendWriteTextAsync(file, contents, false).AsAsyncAction();
+
+		public static IAsyncAction AppendTextAsync(IStorageFile file, string contents) => AppendWriteTextAsync(file, contents, true).AsAsyncAction();
+
+		public static IAsyncAction WriteLinesAsync(IStorageFile file, IEnumerable<string> lines)
+			=> WriteTextAsync(file, ConvertLinesToString(lines));
+		public static IAsyncAction AppendLinesAsync(IStorageFile file, IEnumerable<string> lines)
+			=> AppendTextAsync(file, ConvertLinesToString(lines));
+
+
+		private static async Task<string> ReadTextTaskAsync(IStorageFile file, Streams.UnicodeEncoding encoding)
+		{
+			if (file is null)
+			{
+				// UWP throws NullReferenceException
+				// but it was choosen to throw ArgumentNullException , as it is more appriopriate in this caase
+				throw new ArgumentNullException("StorageFile cannot be null");
+			}
+
+			Encoding encodingForReader = Encoding.UTF8;
+			switch (encoding)
+			{
+				case Streams.UnicodeEncoding.Utf8:
+					encodingForReader = Encoding.UTF8;
+					break;
+				case Streams.UnicodeEncoding.Utf16LE:
+					encodingForReader = Encoding.Unicode;
+					break;
+				case Streams.UnicodeEncoding.Utf16BE:
+					encodingForReader = Encoding.BigEndianUnicode;
+					break;
+			}
+
+			string output = "";
+			using (Stream fileStream = await file.OpenStreamForReadAsync())
+			{
+				using (StreamReader streamReader = new StreamReader(fileStream, encodingForReader))
+				{
+					output = streamReader.ReadToEnd();

```

Comment:
No need for a variable here

```suggestion
					return streamReader.ReadToEnd();

```